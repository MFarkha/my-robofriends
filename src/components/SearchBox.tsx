import { ChangeEventHandler } from "react";

type SearchBoxProps = {
    searchChange: ChangeEventHandler<HTMLInputElement>
};

const SearchBox = ({ searchChange }: SearchBoxProps) => {
    return (
        <div className="pa2">
            <input 
                aria-label="search robots"
                className="pa3 ba b--green bg-lightest-blue" 
                type='search' 
                name="searchBox"
                placeholder="search robots"
                onChange={searchChange}
            />
        </div>
    );
}

export default SearchBox;