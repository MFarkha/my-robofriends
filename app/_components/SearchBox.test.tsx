import SearchBox from "./SearchBox";
import { expect, test, vi } from "vitest";
import { render, screen } from "@testing-library/react";

test("expects to render the SearchBox component", () => {
  const mockSearchChange = vi.fn();
  const { container } = render(<SearchBox searchChange={mockSearchChange} />);
  expect(container).toMatchSnapshot();
});
