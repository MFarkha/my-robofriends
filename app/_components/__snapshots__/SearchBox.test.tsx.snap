// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expects to render the SearchBox component 1`] = `
<div>
  <div
    class="pa2"
  >
    <input
      aria-label="search robots"
      class="pa3 ba b--green bg-lightest-blue"
      name="searchBox"
      placeholder="search robots"
      type="search"
    />
  </div>
</div>
`;
